src/main/java/net/coderbot/iris/UpdateChecker.java

```diff
@@ -0,0 +1,120 @@
+package net.coderbot.iris;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import net.coderbot.iris.config.IrisConfig;
+import net.coderbot.iris.gl.shader.StandardMacros;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.client.Minecraft;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.time.DateUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class UpdateChecker {
+	private final int simpleVersion;
+	private CompletableFuture<UpdateInfo> info;
+	private boolean shouldShowUpdateMessage;
+	private boolean usedIrisInstaller;
+
+	public UpdateChecker(int simpleVersion) {
+		this.simpleVersion = simpleVersion;
+		if (Objects.equals(System.getProperty("iris.installer", "false"), "true")) {
+			usedIrisInstaller = true;
+		}
+	}
+
+	public void checkForUpdates(IrisConfig irisConfig) {
+		this.info = CompletableFuture.supplyAsync(() -> {
+			if (!irisConfig.shouldDisableUpdateMessage()) {
```

Comment:
You can hoist this out of the future, just return from the checkForUpdates method if update checking is disabled.